app/controllers/shipments_controller.rb

```diff
@@ -1,5 +1,18 @@
 class ShipmentsController < ApplicationController
+
+  rescue_from ActiveRecord::RecordNotFound, with: :not_found
+
   def index
     @shipments = Shipment.all
   end
+
+  def show
+    @company = Company.find_by!(id: params[:company_id])
+    @shipment = @company.shipments.find_by!(id: params[:id])
```

Comment:
Usually, we will just use find
```suggestion
    @shipment = @company.shipments.find(id: params[:id])
```

Reply:
Good point that you noticed this should be limited to one company shipments, how about `index` action?

Reply:
fix these two on task 4, but I should've done this earlier.. thanks for your suggestion!